File: C-Sharp/Desafio03/Desafio03/com.gft.model/Carro.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Desafio03.com.gft.model
{
    class Carro:Veiculo
    {
        private int portas;
        private int ano;

        public int Portas { get => portas; set => portas = value; }
        public int Ano { get => ano; set => ano = value; }

		public override void abastecer(int qtd)
		{
			qtd = LitrosCombustivel + qtd;
			if (qtd > 100)
			{
				Console.WriteLine("Seu tanque só comporta 100 litros");
				LitrosCombustivel = 100;
			}
			else
			{
				LitrosCombustivel = qtd;
			}
			Console.WriteLine("Litros de combustível após abastecer: " + LitrosCombustivel + " litros");
		}

		public override void acelerar()
		{
			//verificar se está ligado
			if (IsLigado == true)
			{
				Velocidade = Velocidade + 20;
				LitrosCombustivel = LitrosCombustivel - 1;
				Console.WriteLine("Velocidade após aceleração: " + Velocidade + " km/h" + "\nLitros de combustível após aceleração: " + LitrosCombustivel + " litros");
			}
			else
			{
				Console.WriteLine("Não é possível acelerar com o carro desligado");
			}
		}
	}
}

File: Java/Desafio3 - GFT/src/model/Carro.java
package model;

public class Carro extends Veiculo {
	
	private int portas;
	private int ano;
	
	public int getPortas() {
		return portas;
	}
	public void setPortas(int portas) {
		this.portas = portas;
	}
	public int getAno() {
		return ano;
	}
	public void setAno(int ano) {
		this.ano = ano;
	}

	@Override
	public void abastecer(int qtd) {
		qtd = getLitrosCombustivel() + qtd;
		if (qtd > 100) {
			System.out.println("Seu tanque só comporta 100 litros");
			setLitrosCombustivel(100);
		} else {
				setLitrosCombustivel(qtd);
		}
		System.out.println("Litros de combustível após abastecer: " + getLitrosCombustivel() + " litros");
	}
	
	@Override
	public void acelerar() {
		//verificar se está ligado
		if (getLigado() == true) {
			setVelocidade(getVelocidade()+20);
			setLitrosCombustivel(getLitrosCombustivel() - 1);
			System.out.println("Velocidade após aceleração: " + getVelocidade() + " km/h" + "\nLitros de combustível após aceleração: " + getLitrosCombustivel() + " litros");
		} else {
			System.out.println("Não é possível acelerar com o carro desligado");
		}
	}
	
}
